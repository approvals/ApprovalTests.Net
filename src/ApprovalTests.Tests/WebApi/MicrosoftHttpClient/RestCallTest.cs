using System.Net;
using ApprovalTests.WebApi.MicrosoftHttpClient;
using NUnit.Framework;

namespace ApprovalTests.Tests.WebApi.MicrosoftHttpClient
{
    [TestFixture]
    public class RestCallTest
    {
        [Test]
        public void TestGoogleQuery()
        {
            Approvals.Verify(new GoogleQuery("lolcats"));
        }
    }

    public class GoogleQuery : RestQuery<GoogleQueryResults>
    {
        private readonly string searchTerm;

        public GoogleQuery(string searchTerm)
        {
            this.searchTerm = searchTerm;
        }

        public override string GetQuery()
        {
            return $"?v=1.0&q={searchTerm}";
        }

        public override string GetBaseAddress()
        {
            return "http://ajax.googleapis.com/ajax/services/search/web";
        }

        public override GoogleQueryResults Load()
        {
            return new GoogleQueryResults(GetResponse().Result);
        }
    }

    public class GoogleQueryResults
    {
        public GoogleQueryResults(DownloadStringCompletedEventArgs result)
        {
            //do extraction stuff
        }
    }
}