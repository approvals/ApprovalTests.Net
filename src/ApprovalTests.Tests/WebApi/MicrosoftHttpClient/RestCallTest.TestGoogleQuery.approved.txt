?v=1.0&q=lolcats